webapp/channels/src/components/spinner_button.test.tsx

```diff
@@ -1,76 +1,103 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {mount, shallow} from 'enzyme';
 import React from 'react';
 
 import SpinnerButton from 'components/spinner_button';
 
+import {withIntl} from 'tests/helpers/intl-test-helper';
+import {render, screen, userEvent} from 'tests/react_testing_utils';
+
 describe('components/SpinnerButton', () => {
-    test('should match snapshot with required props', () => {
-        const wrapper = shallow(
-            <SpinnerButton
-                spinning={false}
-                spinningText='Test'
-            />,
+    test('should render button when not spinning', () => {
+        render(
+            withIntl(
+                <SpinnerButton
+                    spinning={false}
+                    spinningText='Test'
+                >
+                    {'Click me'}
+                </SpinnerButton>,
+            ),
         );
-        expect(wrapper).toMatchSnapshot();
+
+        const button = screen.getByRole('button', {name: 'Click me'});
+        expect(button).toBeInTheDocument();
+        expect(button).toBeEnabled();
     });
 
-    test('should match snapshot with spinning', () => {
-        const wrapper = shallow(
-            <SpinnerButton
-                spinning={true}
-                spinningText='Test'
-            />,
+    test('should show spinning text when spinning', () => {
+        render(
+            withIntl(
+                <SpinnerButton
+                    spinning={true}
+                    spinningText='Loading...'
+                />,
+            ),
         );
-        expect(wrapper).toMatchSnapshot();
+
+        const button = screen.getByRole('button');
+        expect(button).toBeInTheDocument();
+        expect(screen.getByText('Loading...')).toBeInTheDocument();
```

Comment:
Any way we can assert the spinner is actually spinning? (I guess, the spinner icon is being rendered)